File: src/main/java/com/github/frcsty/frozenactions/FrozenActions.java
package com.github.frcsty.frozenactions;

import com.github.frcsty.frozenactions.actions.Action;
import com.github.frcsty.frozenactions.actions.ActionContext;
import com.github.frcsty.frozenactions.wrapper.ActionHandler;
import me.mattstudios.msg.base.MessageOptions;
import me.mattstudios.msg.base.internal.Format;
import org.bukkit.World;
import org.bukkit.plugin.java.JavaPlugin;
import org.jetbrains.annotations.NotNull;

public final class FrozenActions extends JavaPlugin {

    @Override
    public void onEnable() {
        final ActionHandler handler = new ActionHandler(this);

        handler.loadDefaults(false);

        handler.createAdventureMessage(MessageOptions.builder()
                .removeFormat(Format.ITALIC)
                .build()
        );

        handler.setAction(new Action() {
            @Override
            public @NotNull String getId() {
                return "time";
            }

            @Override
            public void run(final @NotNull ActionContext context) {
                final World world = context.getPlayer().getWorld();

                world.setTime(context.getDataAsInt());
            }
        });

        handler.removeAction("time");
    }

}

File: src/main/kotlin/com/github/frcsty/frozenactions/FrozenActions.kt
package com.github.frcsty.frozenactions

import com.github.frcsty.frozenactions.wrapper.ActionHandler
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.plugin.java.JavaPlugin

class FrozenActions : JavaPlugin(), Listener {

    private val handler = ActionHandler(this)

    override fun onEnable() {
        handler.loadDefault()

        this.server.pluginManager.registerEvents(this, this)
    }

    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        handler.execute(event.player, listOf("[BROADCAST] Test Action"))
    }
}